src/mbgl/map/transform.cpp

```diff
@@ -198,9 +196,9 @@ void Transform::flyTo(const CameraOptions &camera, const AnimationOptions &anima
 
     /// w₀: Initial visible span, measured in pixels at the initial scale.
     /// Known henceforth as a <i>screenful</i>.
-    double w0 = padding ? std::max(state.size.width, state.size.height)
-                        : std::max(state.size.width - padding.left - padding.right,
-                                   state.size.height - padding.top - padding.bottom);
+    double w0 = padding ? std::max(state.size.width - padding->left - padding->right,
+                                   state.size.height - padding->top - padding->bottom)
+                        : std::max(state.size.width, state.size.height);
```

Comment:
Wow, great catch! I wonder if this makes #8300 more difficult to hit. Hard to say just from inspecting the code, since `r()` is a bit convoluted.